tests/components/qbus/test_config_flow.py

```diff
@@ -0,0 +1,249 @@
+"""Test config flow."""
+
+import json
+import time
+from unittest.mock import patch
+
+from qbusmqttapi.discovery import QbusDiscovery
+
+from homeassistant.components.qbus.const import CONF_ID, CONF_SERIAL_NUMBER, DOMAIN
+from homeassistant.components.qbus.coordinator import QbusConfigCoordinator
+from homeassistant.config_entries import SOURCE_MQTT, SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.service_info.mqtt import MqttServiceInfo
+
+from .common import PAYLOAD_CONFIG, TOPIC_CONFIG
+
+
+async def test_step_mqtt_empty_payload(hass: HomeAssistant) -> None:
+    """Test mqtt discovery with empty payload."""
+    discovery = MqttServiceInfo(
+        subscribed_topic="",
+        topic="",
+        payload=b"",
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_MQTT}, data=discovery
+    )
+
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "invalid_discovery_info"
+
+
+async def test_step_mqtt_invalid_topic(hass: HomeAssistant) -> None:
+    """Test mqtt discovery with an invalid topic."""
+    discovery = MqttServiceInfo(
+        subscribed_topic="invalid/topic",
+        topic="",
+        payload=b"{}",
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_MQTT}, data=discovery
+    )
+
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "invalid_discovery_info"
+
+
+async def test_handle_gateway_topic_when_online(hass: HomeAssistant) -> None:
+    """Test handling of gateway topic with payload indicating online."""
+    discovery = MqttServiceInfo(
+        subscribed_topic="cloudapp/QBUSMQTTGW/state",
+        topic="cloudapp/QBUSMQTTGW/state",
+        payload='{ "online": true }',
+        qos=0,
+        retain=False,
+        timestamp=time.time(),
+    )
+
+    with (
+        patch("homeassistant.components.mqtt.client.async_publish") as mock_publish,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_MQTT}, data=discovery
+        )
+
+    assert mock_publish.called
+    assert result.get("type") == FlowResultType.ABORT
+    assert result.get("reason") == "discovery_in_progress"
+
+
+async def test_handle_gateway_topic_when_offline(hass: HomeAssistant) -> None:
```

Comment:
this can be merged with the test above by parameterizing `payload` and the boolean for `mock_published.called`